boards/arm/stm32g0316_disco/pinmux.c

```diff
@@ -15,6 +15,12 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PA9, STM32G0_PINMUX_FUNC_PA9_USART1_TX},
 	{STM32_PIN_PB7, STM32G0_PINMUX_FUNC_PB7_USART1_RX},
 #endif
+
+#ifdef CONFIG_ADC_1
+    {STM32_PIN_PA1, STM32_MODER_ANALOG_MODE},
+    {STM32_PIN_PA5, STM32_MODER_ANALOG_MODE},
+    {STM32_PIN_PA6, STM32_MODER_ANALOG_MODE},
+#endif /* CONFIG_ADC_1 */
```

Comment:
This should use new DT macros and the defines for the mode should also be adjusted to the Zephyr style, see [here](https://github.com/zephyrproject-rtos/zephyr/blob/ab3e778f47d8460f046967c915874ec7137f636a/boards/arm/nucleo_l073rz/pinmux.c#L34) for an example.

You have to add the analog input defines for G0 series [here](https://github.com/tracychen1/zephyr/tree/add-adc-stm32g0/drivers/pinmux/stm32) first. See other MCU series for examples.